### PR TITLE
Refactored mx_Dialog_cancelButton to replace ugly style rule

### PR DESCRIPTION
**Pull request by JeanPaulLucien**
I found this in styles:
```
.mx_Dialog_cancelButton {
    background-color: #9fa9ba;
    cursor: pointer;
    height:unset;height:18px;-webkit-mask:url(../../img/cancel.12c5c12.svg);
    mask:url(../../img/cancel.12c5c12.svg);
    -webkit-mask-position:center;mask-position:center;-webkit-mask-repeat:no-

repeat;mask-repeat:no-repeat;-webkit-mask-size:cover;mask-size:cover;position:unset;position:absolute;
    right:0;top:10px;
    width:unset;
    width:18px;
}
```
I did not find an element with with class. I hope it will be okay.

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

type: enhancement
notes: better performance with new "Cancel" icon for dialog windows
Result: 
image
How it looks in my Element Web:
image

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * better performance with new "Cancel" icon for dialog windows ([\#10556](https://github.com/matrix-org/matrix-react-sdk/pull/10556)). Contributed by @JeanPaulLucien.

### CODE REVIEW

File: res/css/_common.pcss [hunk 1]
```diff
@@ -481,13 +481,43 @@ legend {
     height: unset;
 }
 
-.mx_Dialog_cancelButton {
-    @mixin customisedCancelButton;
```
Comment (luixxiul) [inline res/css/_common.pcss:485]: `@mixin` is a legitimate Sass function, and currently there should not be a plan to replace them as far as I know.

File: res/css/_common.pcss [hunk 2]
```diff
@@ -481,13 +481,43 @@ legend {
...
-    height: 18px;
+.mx_Dialog_headerWithCancel div.mx_Dialog_cancelButton {
+  width: 30px;
+  height: 30px;
+  transform: rotate(45deg);
+  position: absolute;
+  top: 10px;
+  right: 0; 
+  
+  &.mx_Dialog_cancelButton {
```
Comment (luixxiul) [inline res/css/_common.pcss:492]: This outputs `.mx_Dialog_headerWithCancel div.mx_Dialog_cancelButton.mx_Dialog_cancelButton`, which does not work.

File: res/css/_common.pcss [hunk 3]
```diff
@@ -481,13 +481,43 @@ legend {
...
+    &:before {
+      width: 4px;
+      height: 26px;
+      top: 2px;
+      left: 13px;
+    }
+  }
+  
+  -webkit-border-radius: 2px;
+  -moz-border-radius: 2px;
```
Comment (luixxiul) [inline res/css/_common.pcss:510]: Declarations with prefixes are generated automatically when compiled, so you would not need to add ones manually.

**Comment by luixxiul** [comment]
The component with the class name has been used on several places like `src/components/structures/auth/forgot-password/VerifyEmailModal.tsx`. Since style rules on `res/css/_common.pcss` tend to be critical, I would like to suggest you to run the UI locally to test your changes to make sure whether they fix an issue as expected.

Just my two cents.

File: res/css/_common.pcss (hunk 3 above)
Comment (JeanPaulLucien) [reply res/css/_common.pcss:510]: In fact, because your support browser are the 2 latest versions, you may drop to generate.

**Comment by JeanPaulLucien** [comment]
> The component with the class name has been used on several places like `src/components/structures/auth/forgot-password/VerifyEmailModal.tsx`

**Comment by JeanPaulLucien** [comment]
Oops.

File: res/css/_common.pcss (hunk 1 above)
Comment (luixxiul) [reply res/css/_common.pcss:485]: That mixin is used to make the style of the cancel button on dialogs consistent. Please search the repository for `@mixin customisedCancelButton`, and you will find there are six other lines which use that mixin, such as on `_CompoundDialog.pcss`.

Like @alunturner, it is not really clear to me too what the goal of this PR is.

**Comment by luixxiul** [comment]
I suspect that result might have been taken from browser's devtool.

**Comment by JeanPaulLucien** [comment]
Result is screenshot where I used black background for more contrast.

**Comment by dbkr** [comment]
I'm also not really sure what the intention is here. Why is it better to not use the mixin? Why is the width being changed if it's just a refactor? Why all the extra properties too? What's with the hardcoded colour values?

**Comment by dbkr** [comment]
Also if this is really just a refactor, this would be a Task rather than a Defect.

**Comment by JeanPaulLucien** [comment]
> this would be a Task rather than a Defect.

Then PR about T-Enhancement, because "boost performance" and "user-facing".

**Comment by dbkr** [comment]
So is this changing the design or not? If so, it's not really accurate to call it a refactor, and also the PR would need both before and after screenshots.

**Pull request closed without merging**